### PR TITLE
:read-cond :allow for reading during tty transport

### PR DESCRIPTION
**Pull request by JasonKDarby**
Fixes https://github.com/nrepl/nrepl/issues/272

Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [x] You've updated the [changelog](../CHANGELOG.md)(that only applies to user-visible changes)

Thanks!

*If you're just starting out to hack on nREPL you might find this [section of its
manual][1] extremely useful.*

[1]: https:/nrepl.org/nrepl/hacking_on_nrepl.html

### CODE REVIEW

File: src/clojure/nrepl/transport.clj [hunk 1]
```diff
@@ -164,6 +164,14 @@
             (.close in)
             (.close out))))))))
 
+(defn tty-read-msg
```
Comment (JasonKDarby) [inline src/clojure/nrepl/transport.clj:167]: I really don't like returning a function from this, if you have suggestions I'd appreciate it.

Comment (bbatsov) [reply src/clojure/nrepl/transport.clj:167]: You can always keep it as a lambda and do some integration test of the transport instead. I guess the only reason you made this a public function is to be able to write that unit test for it.

Comment (JasonKDarby) [reply src/clojure/nrepl/transport.clj:167]: Yeah, I did it so I could demonstrate the change was very focused.  I'll put it back.

File: test/clojure/nrepl/transport_test.clj [hunk 2]
```diff
@@ -9,3 +9,15 @@
...
+
+(deftest tty-read-conditional-test
+  (testing "tty-read-msg is configured to read conditionals"
+    (let [in (-> "(try nil (catch #?(:clj Throwable :cljr Exception) e nil))"
+                 (java.io.StringReader.)
+                 (java.io.PushbackReader.))
+          out (ByteArrayOutputStream.)]
+      (is (= ['(try nil (catch Throwable e nil))]
+             (let [FnTransport (sut/tty in out nil)]
+               (.recv FnTransport)     ;; :op "clone"
```
Comment (JasonKDarby) [inline test/clojure/nrepl/transport_test.clj:21]: @bbatsov this wasn't as bad as I originally thought.
I saw eastwood fails on `.recv` saying "reference to field recv can't be resolved".  Any tip on how to resolve this?

Comment (bbatsov) [reply test/clojure/nrepl/transport_test.clj:21]: Probably you need to add a type signature there. Btw, I think the variable should be named `fn-transport`, not `FnTransport` as this looks like some type name. //cc @vemv

Comment (JasonKDarby) [reply test/clojure/nrepl/transport_test.clj]: There's a type `FnTransport` and also a defn `fn-transport` in nrepl.transport.  There must be a better name but I just named it `fn-transport` as suggested.

**Comment by bbatsov** [comment]
Thanks for tackling this!

**Pull request merged**